dpnp/backend/kernels/dpnp_krnl_common.cpp

```diff
@@ -810,9 +809,12 @@ DPCTLSyclEventRef dpnp_matmul_c(DPCTLSyclQueueRef q_ref,
     std::vector<sycl::event> dep_events = cast_event_vector(dep_event_vec_ref);
     sycl::event event;
 
-    _DataType* array_1 = reinterpret_cast<_DataType*>(const_cast<void*>(input1_in));
-    _DataType* array_2 = reinterpret_cast<_DataType*>(const_cast<void*>(input2_in));
-    _DataType* result = reinterpret_cast<_DataType*>(result_out);
+    DPNPC_ptr_adapter<_DataType> input1_ptr(q, input1_in, size_m * size_k);
+    DPNPC_ptr_adapter<_DataType> input2_ptr(q, input2_in, size_k * size_n);
+    DPNPC_ptr_adapter<_DataType> result_ptr(q, result_out, size_m * size_n, false, true);
+    _DataType* array_1 = input1_ptr.get_ptr();
+    _DataType* array_2 = input2_ptr.get_ptr();
+    _DataType* result = result_ptr.get_ptr();
```

Comment:
Do we really need `DPNPC_ptr_adapter` here? What is purpose of these changes?

Reply:
I think it's been changed like everywhere else. I wouldn't like to patch PR and wait for CI. Would you like to?